wsmaster/che-core-api-auth-bitbucket/src/main/java/org/eclipse/che/security/oauth/BitbucketOAuthAuthenticator.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2012-2022 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.security.oauth;
+
+import com.google.api.client.util.store.MemoryDataStoreFactory;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import javax.inject.Singleton;
+import org.eclipse.che.api.auth.shared.dto.OAuthToken;
+import org.eclipse.che.commons.json.JsonHelper;
+import org.eclipse.che.commons.json.JsonParseException;
+import org.eclipse.che.security.oauth.shared.User;
+
+/** OAuth authentication for BitBucket SAAS account. */
+@Singleton
+public class BitbucketOAuthAuthenticator extends OAuthAuthenticator {
+  public BitbucketOAuthAuthenticator(
+      String clientId, String clientSecret, String[] redirectUris, String authUri, String tokenUri)
+      throws IOException {
+    configure(
+        clientId, clientSecret, redirectUris, authUri, tokenUri, new MemoryDataStoreFactory());
+  }
+
+  @Override
+  public User getUser(OAuthToken accessToken) throws OAuthAuthenticationException {
+    BitbucketUser user =
+        getJson("https://api.bitbucket.org/2.0/user", accessToken.getToken(), BitbucketUser.class);
```

Comment:
Maybe it's better to have a constant for `https://api.bitbucket.org/2.0/user`

Reply:
done